vllm-ext/inference_engine/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+#  Copyright (c) 2023 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+cmake_minimum_required(VERSION 3.12) # Don't bump this version for no reason
+project("inference_engine" C CXX)
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+if (NOT XCODE AND NOT MSVC AND NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE Release CACHE STRING "Build type" FORCE)
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
+
+#
+# Option list
+#
+
+# general
+option(IE_STATIC                 "inference_engine: static link libraries"                          OFF)
+option(IE_NATIVE                 "inference_engine: enable -march=native flag"                      OFF)
+option(IE_LTO                    "inference_engine: enable link time optimization"                  OFF)
+
+# debug
+option(IE_ALL_WARNINGS           "inference_engine: enable all compiler warnings"                   ON)
+option(IE_ALL_WARNINGS_3RD_PARTY "inference_engine: enable all compiler warnings in 3rd party libs" OFF)
+option(IE_GPROF                  "inference_engine: enable gprof"                                   OFF)
+
+# sanitizers
+option(IE_SANITIZE_THREAD        "inference_engine: enable thread sanitizer"                        OFF)
+option(IE_SANITIZE_ADDRESS       "inference_engine: enable address sanitizer"                       OFF)
+option(IE_SANITIZE_UNDEFINED     "inference_engine: enable undefined sanitizer"                     OFF)
+
+# instruction set specific
+option(IE_AVX                    "inference_engine: enable AVX"                                     ON)
+option(IE_AVX2                   "inference_engine: enable AVX2"                                    ON)
+option(IE_F16C                   "inference_engine: enable F16C"                                    ON)
+option(IE_AVX512                 "inference_engine: enable AVX512"                                  OFF)
```

Comment:
Do we build AVX512 by default? I am not sure the usage of the option here.

Reply:
It's actually not used for now. Instrinsics are checked dynamically during run-time. Will keep it here for later.